heudiconv/tests/test_archives.py

```diff
@@ -0,0 +1,90 @@
+from glob import glob
+import os
+import os.path as op
+from pathlib import Path
+import stat
+import typing
+
+
+import pytest
+import shutil
+from .utils import TESTS_DATA_PATH
+from ..parser import get_extracted_dicoms
+
+
+@pytest.fixture
+def get_dicoms_gztar(tmpdir: Path) -> typing.List[str]:
+    tmp_file = tmpdir / "dicom"
+    archive = shutil.make_archive(
+        str(tmp_file),
+        format="gztar", 
+        root_dir=TESTS_DATA_PATH, 
+        base_dir="01-anat-scout")
+    return [archive]
+
+
+@pytest.fixture
+def get_dicoms_zip(tmpdir: Path) -> typing.List[str]:
+    tmp_file = tmpdir / "dicom"
+    archive = shutil.make_archive(
+        str(tmp_file),
+        format="zip", 
+        root_dir=TESTS_DATA_PATH, 
+        base_dir="01-anat-scout")
+    return [archive]
```

Comment:
Sorry -- I have too strong allergy to duplication... what about

```suggestion
def get_dicoms_archive(format: str, tmpdir: Path) -> typing.List[str]:
    tmp_file = tmpdir / "dicom"
    archive = shutil.make_archive(
        str(tmp_file),
        format=format, 
        root_dir=TESTS_DATA_PATH, 
        base_dir="01-anat-scout")
    return [archive]

@pytest.fixture
def get_dicoms_gztar(tmpdir: Path) -> typing.List[str]:
    return get_dicoms_archive("gztar", tmpdir)

@pytest.fixture
def get_dicoms_zip(tmpdir: Path) -> typing.List[str]:
    return get_dicoms_archive("zip", tmpdir)
```

FWIW found some HOWTOs on parametric fixtures in https://stackoverflow.com/questions/42228895/how-to-parametrize-a-pytest-fixture .  Since tests below seems to do exactly that -- could be RFed to use one of those approaches or just become `@pytest.mark.parametrize(`'ed and use suggested above `get_dicoms_archive` directly.

Reply:
First time I've seen parameterized fixtures, but that makes sense to me. Note that I've kept the `get_dicoms_gztar` to allow for a test that mixes archived and non-archived files 

https://github.com/nipy/heudiconv/blob/99b0bf6109732c365721250fd7843c70a74002c1/heudiconv/tests/test_archives.py#L75-L77